### PR TITLE
TMA-816: Make .execute_mufs work with symbolized hashes

### PR DESCRIPTION
**Pull request by panjan**

### CODE REVIEW

File: lib/gooddata/models/user_filters/user_filter_builder.rb [hunk 1]
```diff
@@ -542,7 +543,7 @@ def self.normalize_filters(filters)
     def self.sanitize_filters_to_delete(to_delete, users_brick_input, project_users)
       return to_delete unless users_brick_input && users_brick_input.any?
       user_profiles = users_brick_input.map do |user|
-        result = project_users.find { |u| u.login == user['login'] }
+        result = project_users.find { |u| u.login == user.with_indifferent_access['login'] }
```
Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Contains iterators nested 2 deep (NestedIterators)

Comment (yenkins) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: Has the variable name 'u' (UncommunicativeVariableName)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/754/ : SUCCESS in 27s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/614/ : SUCCESS in 2m 15s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/875/ : SUCCESS in 2m 12s

**Comment by kubamahnert** [comment]
extended test

File: lib/gooddata/models/user_filters/user_filter_builder.rb (hunk 1 above)
Comment (kubamahnert) [inline lib/gooddata/models/user_filters/user_filter_builder.rb]: this is not preventing us from making a similar mistake in the future. How about we turn all results of the realize() call into hashes with indifferent access, so the access will be uniform for reading all sources (CSV, ADS, S3..) ?

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/756/ : SUCCESS in 9s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1211/ : SUCCESS in 56m 06s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/970/ : SUCCESS in 36m 04s

File: lib/gooddata/models/user_filters/user_filter_builder.rb (hunk 1 above)
Comment (panjan) [reply lib/gooddata/models/user_filters/user_filter_builder.rb]: `#realize` does not return a hash. It returns a file path which is then read in inconsistent ways. Created a ticket for fixing the inconsistencies. https://jira.intgdc.com/browse/TMA-838

**Comment by kubamahnert** [comment]
ok to merge

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/284/ : SUCCESS in 17s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/617/ : SUCCESS in 50s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/878/ : SUCCESS in 34s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1212/ : SUCCESS in 55m 07s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/971/ : SUCCESS in 45m 25s

**Pull request merged**